### PR TITLE
client: use tokio channels

### PR DESCRIPTION
**Pull request by niklasad1**
This PR replaces the future channels with tokio because the APIs fit our use-cases better.

In addition #962 will migrate to tokio channels as well.

### CODE REVIEW

File: core/src/client/async_client/mod.rs [hunk 1]
```diff
@@ -765,21 +768,27 @@ async fn background_task<S, R>(
...
-				message_fut = next_message_fut;
+				message_fut = msg;
+				exit_or_ping_fut = future::select(on_exit, ping_fut(ping_interval));
 			}
 		};
 	}
 
 	// Wake the `on_disconnect` method.
-	let _ = on_close.send(());
+	let _ = frontend.close();
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:791]:
```suggestion
	_ = frontend.close();
```

File: core/src/client/async_client/mod.rs [hunk 2]
```diff
@@ -765,21 +768,27 @@ async fn background_task<S, R>(
...
+				exit_or_ping_fut = future::select(on_exit, ping_fut(ping_interval));
 			}
 		};
 	}
 
 	// Wake the `on_disconnect` method.
-	let _ = on_close.send(());
+	_ = frontend.close();
 	// Send close message to the server.
 	let _ = sender.close().await;
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:793]:
```suggestion
	_ = sender.close().await;
```

File: core/src/client/mod.rs [hunk 3]
```diff
@@ -256,10 +255,10 @@ impl<Notif> Subscription<Notif> {
 			SubscriptionKind::Method(notif) => FrontToBack::UnregisterNotification(notif),
 			SubscriptionKind::Subscription(sub_id) => FrontToBack::SubscriptionClosed(sub_id),
 		};
-		self.to_back.send(msg).await?;
+		self.to_back.send(msg).await.unwrap();
```
Comment (jsdw) [inline core/src/client/mod.rs:258]: When would this panic?

**Review by jsdw — APPROVED** [review]
LGTM offhand; just one `unwrap` that I wasn't sure on!

File: core/src/client/mod.rs (hunk 3 above)
Comment (niklasad1) [reply core/src/client/mod.rs:258]: oops sorry :)

**Review by lexnv — APPROVED** [review]

**Pull request merged**